weatherman/calculations.py

```diff
@@ -0,0 +1,146 @@
+                    if max_temp_value > max_temp_tuple[0]:
+
+                        max_temp_tuple = (max_temp_value, weather_entry['PKT'])
+                except ValueError:
+                    pass
+
+            if weather_entry.get('Min TemperatureC'):
+
+                try:
+                    min_temp_value = float(weather_entry['Min TemperatureC'])
+                    if min_temp_value < min_temp_tuple[0]:
+
+                        min_temp_tuple = (min_temp_value, weather_entry['PKT'])
+                except ValueError:
+                    pass
+
+            if weather_entry.get('Max Humidity'):
+
+                try:
+                    max_humid_value = float(weather_entry['Max Humidity'])
+                    if max_humid_value > max_humid_tuple[0]:
+
+                        max_humid_tuple = (max_humid_value, weather_entry['PKT'])
+
+                except ValueError:
+                    continue
+
+        return YearlyExtremeWeatherResult(
+            max_temp_tuple[0],
+            max_temp_tuple[1],
+            min_temp_tuple[0],
+            min_temp_tuple[1],
+            max_humid_tuple[0],
+            max_humid_tuple[1]
+        )
+
+    def compute_monthly_averages_weather(self, month: str) -> MonthlyAveragesWeatherResult:
+        weather_data_filtered_by_month = self.filter_weather_data_by_month(month)
+        sums, counts = defaultdict(float), defaultdict(int)
+        for weather_entry in weather_data_filtered_by_month:
+            try:
+                sums['highest'] += float(weather_entry.get('Max TemperatureC', 0))
+                counts['highest'] += 1
+            except ValueError:
+                pass
+            try:
+                sums['lowest'] += float(weather_entry.get('Min TemperatureC', 0))
+                counts['lowest'] += 1
+            except ValueError:
+                pass
+            try:
+                sums['mean_humid'] += float(weather_entry.get('Mean Humidity', 0))
+                counts['mean_humid'] += 1
+            except ValueError:
+                continue
+
+        avg_highest = sums['highest'] / counts['highest'] if counts['highest'] > 0 else None
+        avg_lowest = sums['lowest'] / counts['lowest'] if counts['lowest'] > 0 else None
+        avg_mean_humid = sums['mean_humid'] / counts['mean_humid'] if counts['mean_humid'] > 0 else None
+
+        return MonthlyAveragesWeatherResult(
+            round(avg_highest, 2),
+            round(avg_lowest, 2),
+            round(avg_mean_humid, 2)
+        )
+
+    def compute_monthly_extreme_weather(self, month: str) -> MonthlyExtremeBarChartResult:
+        weather_data_by_month = self.filter_weather_data_by_month(month)
+        highest_temp: List[float] = []
+        lowest_temp: List[float] = []
+        days: List[str] = []
+        for weather_entry in weather_data_by_month:
+            days.append(weather_entry.get('PKT', ''))
+            try:
```

Comment:
i can see a lot of places where you have used try except, please use a generic and non-repetitive way for this